crates/miden-objects/src/block/block_inputs.rs

```diff
@@ -52,7 +52,7 @@ impl BlockInputs {
     }
 
     /// Returns a reference to the previous block header.
-    pub fn prev_block_header(&self) -> &BlockHeader {
+    pub fn prev_block_header(&self) -> &UnsignedBlockHeader {
         &self.prev_block_header
```

Comment:
This is an example where I find `Deref` surprising. We return a reference to a field of type `BlockHeader` but it magically becomes `&UnsignedBlockHeader` and it's not immediately clear here why it does.